tests/vector_processing/test_rasterize_vector.py

```diff
@@ -0,0 +1,258 @@
+from contextlib import nullcontext
+from typing import ContextManager, NamedTuple, Optional, Union
+
+import geopandas as gpd
+import numpy as np
+import pytest
+from rasterio import profiles, transform
+from shapely.geometry import LineString, Point, box
+
+from eis_toolkit import exceptions
+from eis_toolkit.vector_processing.rasterize_vector import rasterize_vector
+
+SAMPLE_LINE_GEODATAFRAME = gpd.GeoDataFrame(
+    {
+        "geometry": [
+            LineString([(0, 0), (1, 1)]),
+            LineString([(0, 1), (1, 2)]),
+            LineString([(0, 2), (1, 3)]),
+        ],
+        "values": [1, 2, 3],
+    },
+)
+
+SAMPLE_POINT_GEODATAFRAME = gpd.GeoDataFrame(
+    {
+        "geometry": [
+            Point(0, 0),
+            Point(2, 2),
+            Point(3, 3),
+            Point(4, 2),
+            Point(5, 5),
+        ],
+        "values": [1, 2, 3, 4, 5],
+    },
+)
+
+SAMPLE_POLYGON_GEODATAFRAME = gpd.GeoDataFrame(
+    {
+        "geometry": [
+            box(0, 0, 2, 2),
+            Point(5, 5).buffer(2.0),
+            LineString([(6, 6), (10, 10)]).buffer(0.5),
+        ],
+        "values": [1, 2, 3],
+    },
+)
+
+SAMPLE_EMPTY_GEODATAFRAME = gpd.GeoDataFrame(
+    {
+        "geometry": [],
+        "values": [],
+    },
+)
+
+SAMPLE_TRACES_WITH_EMPTY_GEODATAFRAME = gpd.GeoDataFrame(
+    {
+        "geometry": [
+            LineString(),
+            LineString([(0, 1), (1, 2)]),
+            LineString([(0, 2), (1, 3)]),
+        ],
+        "values": [1, 2, 3],
+    },
+)
+
+
+class RasterizeVectorTestArgs(NamedTuple):
+    """Tuple for holding arguments, with defaults, for testing rasterize_vector."""
+
+    geodataframe: gpd.GeoDataFrame
+    resolution: float = 1.0
+    value_column: Optional[str] = None
+    default_value: float = 1.0
+    fill_value: float = 0.0
+    base_raster_profile: Optional[Union[profiles.Profile, dict]] = None
+    buffer_value: Optional[float] = None
+    raises: ContextManager = nullcontext()
+
+
+@pytest.mark.parametrize(
+    "geodataframe,resolution,value_column,default_value,fill_value,base_raster_profile,buffer_value,raises",
+    [
+        pytest.param(
+            *RasterizeVectorTestArgs(geodataframe=SAMPLE_LINE_GEODATAFRAME, resolution=0.05),
```

Comment:
I took some liberties in handling test parametrization. I used a `NamedTuple` to represent the test arguments along with their defaults. This makes refactoring later easier as the order of the test arguments along with their defaults can be only changed in `RasterizeVectorTestArgs` rather than having to do it separately for each tuple of test arguments. Hopefully not too complex...